src/client.py

```diff
@@ -0,0 +1,362 @@
+                # Nested if instead of elif name and self.get_project_id(name=name) to improve performance by reducing
+                # the number of API calls for name or id NoneType check.
+                if p_id:
+                    path = 'projects/' + p_id + '/' + resource + '/'
+        else:
+            if id:
+                path = 'projects/' + id + '/'
+            elif name:
+                p_id = self.get_project_id(name=name)
+                if p_id:
+                    path = 'projects/' + p_id + '/'
+
+        # API call and filters check.
+        if path:
+            if filter and filter_val:
+                payload = {filter: filter_val}
+                return self.api_root(path, parameters=payload)
+            elif filter:
+                return self.api_root(path, parameters=filter)
+            else:
+                return self.api_root(path)
+        return path
+
+    # Get functions.
+    def get_info(self, info):
+        """
+        Gets non-project resource information. Authentication not required for health, version, profiles, projecttags,
+        sites, managements. Authentication required for me.
+        See https://mermaid-api.readthedocs.io/en/latest/nonproject.html for more information on non-project resources.
+        :param info: health, version, profiles, projecttags, sites, managements, me.
+        :type info: str
+        :return: non-project resource data.
+        :rtype: dict, str, None
+        """
+        # TODO: Handle pagination, filters, PUT, HEAD OPTIONS requests
+        # Authentication required for access to 'me' API endpoint.
+        if info == 'me' and not self.authenticated:
+            return None
+
+        if info in self.npr_endpoints:
+            return self.api_root(info)
+        return None
+
+    def get_choices(self):
+        """
+        Gets a list of objects, each one of which has a name item (e.g., countries) and a data item that is a list
+        of available choice objects.
+        :return: list of choice objects.
+        :rtype: list
+        """
+        # TODO: Filter: /updates/?timestamp="DATE"
+        return self.api_root('choices')
+
+    def get_attribute(self, attr):
+        """
+        Gets attributes which are “things that can be observed” – coral and other taxa as well as non-organic
+        benthic substrates for benthic transects and bleaching surveys, fish species/genera/families as well
+        as arbitrary fish groupings for fish belt transects, and so on.
+        :param attr: fishsizes, benthicattributes, fishfamilies, fishgenera, fishspecies, fishgroupings.
+        :type attr: str
+        :return: attribute data.
+        :rtype: dict, None
+        """
+        if attr not in self.attrs_endpoints:
+            return None
+        return self.api_root(attr)
+
+    def get_projects(self, showall=False):
+        """
+        Gets the projects subdirectory. Without query parameters, returns a list of projects of which
```

Comment:
Again not "subdirectory" I think... Also, I'm confused about the difference between this method and `api_projects` above (see renaming suggestion).